server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -248,13 +261,51 @@ public static void validateGeoJson(Map<?, ?> value) {
                 throw new IllegalArgumentException(invalidGeoJSON("geometries field missing"));
             }
 
+            final String[] typeInCollection = {null};
+            final boolean[] sameType = {true};
             forEach(geometries, input -> {
                 if (!(input instanceof Map<?, ?> map)) {
                     throw new IllegalArgumentException(invalidGeoJSON("invalid GeometryCollection"));
                 } else {
-                    validateGeoJson(map);
+                    Map<String, Object> validatedShape = validateGeoJson(map);
+                    String shapeType = BytesRefs.toString(validatedShape.get(TYPE_FIELD));
+                    if (typeInCollection[0] == null) {
+                        typeInCollection[0] = shapeType;
+                    } else {
+                        if (typeInCollection[0].equals(shapeType) == false) {
+                            sameType[0] = false;
+                        }
+                    }
```

Comment:
What about nested collections? there possible converted map must be stored back as well.

Reply:
Thanks for catching, added handling and a test in https://github.com/crate/crate/pull/17012/commits/ed2a9dfcd9a30c8a55d6441c34f71345c0f93ae9